src/alert.cpp

```diff
@@ -839,6 +839,9 @@ namespace libtorrent {
 		, sock_type(t)
 		, address(listen_addr)
 		, port(listen_port)
+#ifndef TORRENT_NO_DEPRECATE
+		, endpoint(tcp::endpoint(listen_addr, listen_port))
```

Comment:
you don't need to construct a temporary object here, you can just pass in the address and port directly to the constructor of endpoint